app/policies/booking_policy.rb

```diff
@@ -0,0 +1,8 @@
+class BookingPolicy < ApplicationPolicy
+  def confirm?
+    user.approved? &&
+      user_owned? &&
+      record.start_time.future? &&
+      record.state == 'provisional'
```

Comment:
I've whitelisted the state which can transition to `confirmed` to ensure we don't let transitions happen if new states get added without updating this policy.